File: snippets/csharp/ListPayments.cs
using Breez.Sdk;

public class ListPaymentsSnippets
{
    public void ListPayments(BlockingBreezServices sdk)
    {
        // ANCHOR: list-payments
        try
        {
            var payments = sdk.ListPayments(
                new ListPaymentsRequest());
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: list-payments
    }

    public void ListPaymentsFiltered(BlockingBreezServices sdk)
    {
        // ANCHOR: list-payments-filtered
        try
        {
            var payments = sdk.ListPayments(
                new ListPaymentsRequest(
                    new() { PaymentTypeFilter.Sent },
                    fromTimestamp: 1696880000,
                    includeFailures: true));
        }
        catch (Exception)
        {
            // Handle error
        }
        // ANCHOR_END: list-payments-filtered
    }
}

File: snippets/python/src/list_payments.py
import breez_sdk
import logging

sdk_services = breez_sdk.BlockingBreezServices


def list_payments():
    try:
        # ANCHOR: list-payments
        req = breez_sdk.ListPaymentsRequest(filters=breez_sdk.PaymentTypeFilter.All)
        sdk_services.list_payments(req=req)
        # ANCHOR_END: list-payments
    except Exception as error:
        logging.error(error)
        raise

def list_payments_filtered():
    try:
        # ANCHOR: list-payments-filtered
        req = breez_sdk.ListPaymentsRequest(
            filters=breez_sdk.PaymentTypeFilter.Sent, 
            from_timestamp=1696880000, 
            include_failures=True)
        sdk_services.list_payments(req=req)
        # ANCHOR_END: list-payments-filtered
    except Exception as error:
        logging.error(error)
        raise